Description: allow some tests to fail on loong64
Author: Bo YU <tsu.yubo@gmail.com>
Last-Update: 2023-12-09
Forwarded: no
--- a/test/parallel/parallel.status
+++ b/test/parallel/parallel.status
@@ -84,6 +84,15 @@
 test-crypto-dh-stateless: SKIP
 test-crypto-keygen: SKIP
 
+[$arch==loong64]
+test-crypto-sign-verify: PASS, FLAKY
+test-freeze-intrinsics: PASS, FLAKY
+test-tls-root-certificates: PASS, FLAKY
+test-vm-global-non-writable-properties: PASS, FLAKY
+test-vm-global-setter: PASS, FLAKY
+test-vm-module-synthetic: PASS, FLAKY
+test-vm-strict-assign: PASS, FLAKY
+
 [$arch==mips64el]
 # the debug flag is for hacking v8 internals
 test-debug-args: PASS,FLAKY
--- a/test/sequential/sequential.status
+++ b/test/sequential/sequential.status
@@ -73,6 +73,10 @@
 # https://github.com/nodejs/node/issues/41286
 test-performance-eventloopdelay: PASS, FLAKY
 
+[$arch==loong64 || $arch==mips64el]
+test-worker-prof: PASS, FLAKY
+test-cpu-prof-dir-relative: PASS, FLAKY
+
 [$arch==riscv64]
 test-diagnostic-dir-cpu-prof: PASS, FLAKY
 test-cpu-prof-worker-argv: PASS, FLAKY
