Description: allow some tests to fail on loong64
Author: Bo YU <tsu.yubo@gmail.com>
Last-Update: 2023-12-09
Forwarded: no
--- a/test/parallel/parallel.status
+++ b/test/parallel/parallel.status
@@ -92,6 +92,15 @@
 # https://github.com/nodejs/node/issues/47297
 test-wasm-web-api: SKIP
 
+[$arch==loong64]
+test-crypto-sign-verify: PASS, FLAKY
+test-freeze-intrinsics: PASS, FLAKY
+test-tls-root-certificates: PASS, FLAKY
+test-vm-global-non-writable-properties: PASS, FLAKY
+test-vm-global-setter: PASS, FLAKY
+test-vm-module-synthetic: PASS, FLAKY
+test-vm-strict-assign: PASS, FLAKY
+
 [$arch==mips64el]
 # the debug flag is for hacking v8 internals
 test-debug-args: PASS,FLAKY
