Description: explicitely load all error strings - w/a bug on mipsel
Forwarded: https://github.com/openssl/openssl/issues/18535
Author: Sebastian Andrzej Siewior <sebastian@breakpoint.cc>
Reviewed-by: Jérémy Lal <kapouer@melix.org>
Last-Update: 2022-06-13
--- a/src/node.cc
+++ b/src/node.cc
@@ -1080,7 +1080,10 @@
         const char* conf = per_process::cli_options->openssl_config.c_str();
         OPENSSL_INIT_set_config_filename(settings, conf);
       }
-      OPENSSL_init_crypto(OPENSSL_INIT_LOAD_CONFIG, settings);
+      OPENSSL_init_crypto(OPENSSL_INIT_LOAD_CONFIG |
+                    OPENSSL_INIT_LOAD_CRYPTO_STRINGS |
+                    OPENSSL_INIT_ADD_ALL_CIPHERS |
+                    OPENSSL_INIT_ADD_ALL_DIGESTS, settings);
       OPENSSL_INIT_free(settings);
 
       if (ERR_peek_error() != 0) {
