Subject: test does not pass on riscv64
Last-Update: 2022-06-13
Author: Jérémy Lal <kapouer@melix.org>
Forwarded: not-yet

--- a/test/sequential/sequential.status
+++ b/test/sequential/sequential.status
@@ -53,3 +53,13 @@
 [$arch==s390x]
 # https://github.com/nodejs/node/issues/41286
 test-performance-eventloopdelay: PASS, FLAKY
+
+[$arch==riscv64]
+test-diagnostic-dir-cpu-prof: FLAKY
+test-cpu-prof-worker-argv: FLAKY
+test-cpu-prof-exit: FLAKY
+test-cpu-prof-kill: FLAKY
+test-diagnostic-dir-cpu-prof: FLAKY
+test-debugger-preserve-breaks: FLAKY
+test-cpu-prof-dir-relative: FLAKY
+
